src/include/storage/varlen_pool.h

```diff
@@ -0,0 +1,78 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Terrier
+//
+// varlen_pool.h
+//
+// Identification: src/include/storage/varlen_pool.h
+//
+// Copyright (c) 2015-2018, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <cstdint>
+#include <cstdlib>
+#include <unordered_set>
+
+#include "common/spin_latch.h"
+
+namespace terrier {
+
+/**
+ * @brief A varlen pool that can quickly allocate chunks of memory.
+ * */
+class VarlenPool {
```

Comment:
Write doxygen docs for every public method

Reply:
Written